packages/hyperion-autologging/src/ALInteractableDOMElement.ts

```diff
@@ -0,0 +1,185 @@
+    const closestSelectorElement = node.closest(selectorString);
+    if (
+      closestSelectorElement == null ||
+      (closestSelectorElement instanceof HTMLElement &&
+        ignoreInteractiveElement(closestSelectorElement))
+    ) {
+      const closestHandler = getClosestHandler(node, eventName);
+      return closestHandler != null
+        ? returnInteractableNode
+          ? closestHandler
+          : node
+        : null;
+    } else {
+      if (closestSelectorElement instanceof HTMLElement) {
+        return returnInteractableNode ? closestSelectorElement : node;
+      }
+    }
+  }
+}
+
+function getClosestHandler(node: EventTarget, eventName: string): HTMLElement | null {
+  if (node instanceof HTMLElement) {
+    let handler;
+    if (eventNamesMap[eventName] != null) {
+      handler = eventNamesMap[eventName](node);
+    }
+    if (handler != null) {
+      return ignoreInteractiveElement(node) ? null : node;
+    } else {
+      return node.parentElement != null
+        ? getClosestHandler(node.parentElement, eventName)
+        : null;
+    }
+  } else {
+    return null;
+  }
+}
+
+function ignoreInteractiveElement(node: HTMLElement) {
+  const innerHeight: number = window.innerHeight;
+  const innerWidth: number = window.innerWidth;
+  return (
+    node.tagName === 'BODY' ||
+    node.tagName === 'HTML' ||
+    (node.clientHeight === innerHeight && node.clientWidth === innerWidth)
+  );
+}
+
+export function trackInteractable(events: Array<string>): void {
+  addEventListener.onArgsObserverAdd(function (
+    this: EventTarget,
+    event,
+    _listener,
+  ) {
+    if (events.indexOf(event) > -1 && this instanceof HTMLElement) {
+      if (!ignoreInteractiveElement(this)) {
+        const attribute = `data-${event}able`;
+
+        this.setAttribute(attribute, '1');
+      }
+    }
+  });
+  removeEventListener.onArgsObserverAdd(function (
+    this: EventTarget,
+    event,
+    _listener,
+  ) {
+    if (this instanceof HTMLElement) {
+      const attribute = `data-${event}able`;
+
+      this.removeAttribute(attribute);
+    }
+  });
+}
+export function trackSynthetic(event: string): void {
+  onReactDOMElement.add(
+    (_component, props: ReactComponentObjectProps) => {
+      if (
+        props != null &&
+        props[SYNTHETIC_MOUSE_EVENT_HANDLER_MAP[event]] != null
+      ) {
+        props[`data-${event}able`] = '1';
+      }
+    },
+  );
+}
+export default function isTruthy(value: any): boolean {
+  return (
+    /* Although Boolean(value) also captures null/undefined we need to add this
+      check so that Flow refines value as non-nullable */
+    value != null && Boolean(value)
+  );
+}
+const extractInnerText = (element: HTMLElement | null): string | null => {
+  const innerText = element?.innerText?.replace(
+    // Remove zero-width invisible Unicode characters (https://stackoverflow.com/a/11305926)
+    /[\u200B-\u200D\uFEFF]/g,
+    '',
+  );
+  if (isTruthy(innerText)) {
```

Comment:
where is this defined?

Reply:
isTruthy is defined right above extractInnerText() on line 148.